api/include/opentelemetry/metrics/sync_instruments.h

```diff
@@ -104,29 +114,33 @@ class UpDownCounter : public SynchronousInstrument
    *
    * @param value The amount of the measurement.
    */
-  virtual void Add(T value) noexcept = 0;
+  virtual void Add(T value, const opentelemetry::context::Context &context) noexcept = 0;
 
   /**
    * Add a value with a set of attributes.
    *
    * @param value The increment amount. May be positive, negative or zero.
    * @param attributes A set of attributes to associate with the count.
    */
-  virtual void Add(T value, const common::KeyValueIterable &attributes) noexcept = 0;
+  virtual void Add(T value,
+                   const common::KeyValueIterable &attributes,
+                   const opentelemetry::context::Context &context) noexcept = 0;
 
   template <class U,
             nostd::enable_if_t<common::detail::is_key_value_iterable<U>::value> * = nullptr>
-  void Add(T value, const U &attributes) noexcept
+  void Add(T value, const U &attributes, const opentelemetry::context::Context &context) noexcept
   {
-    this->Add(value, common::KeyValueIterableView<U>{attributes});
+    this->Add(value, common::KeyValueIterableView<U>{attributes}, context);
   }
 
   void Add(T value,
-           std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>>
-               attributes) noexcept
+           std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>> attributes,
+           const opentelemetry::context::Context &context) noexcept
   {
-    this->Add(value, nostd::span<const std::pair<nostd::string_view, common::AttributeValue>>{
-                         attributes.begin(), attributes.end()});
+    this->Add(value,
+              nostd::span<const std::pair<nostd::string_view, common::AttributeValue>>{
+                  attributes.begin(), attributes.end()},
+              context);
```

Comment:
Should we make `context` optional, and use default runtime context if not provided. 

```cpp
void Add(value);
void Add(value, context);
void Add(value, attributes);
void Add(value, attributes, context);
```

or just

```cpp
void Add(value)
void Add(value, attributes)
void Add(value, attributes, context)
```

Reply:
thanks, added these four:

```cpp
void Add(value);
void Add(value, context);
void Add(value, attributes);
void Add(value, attributes, context);
```